src/main/java/model/Car.java

```diff
@@ -0,0 +1,39 @@
+package model;
+
+import util.validator.NameValidator;
+
+public class Car {
+	private final String name;
+	private int position;
```

Comment:
```suggestion
	private final String name;
	
	private int position;
```

키워드 종류가 달라지면 띄워쓰기로 분류해놓는게 좋아용

Reply:
Car의 경우 필드 변수로 name을 가지고 있네요. name의 검증은 **어디서** 이루어져야 할까? 에 대해 고민해봅시다.
현재 Cars에서도 validateNames로 이름들을 검증하고 있고, Car에서는 validateName을 통해 이름을 검증하고 있습니다.

이를 개선시킬 방법이 있을까요?

힌트로 아래 글을 공유해봅니다!!
[객체지향 생활체조 제 3 원칙: 모든 원시값과 문자열을 포장하라](https://tecoble.techcourse.co.kr/post/2020-05-29-wrap-primitive-type/)

Reply:
_객체지향적인 코드란 무엇인가?_ 에 대해 감이 안 잡힐 수도 있어요. 저 같은 경우엔 이런 식의 사고가 좀 더 방향을 잡아줬습니다.

> 이름이 검증되야 한다는 걸 Car가 알아야 할까?
> Car에게 달리라는 메세지를 보낼 때, 외부에서 인수로 canGo를 주입하는 게 맞는가? 알아서 달릴 필요는 없는가?

좀 더 전문적으로 표현을 해보자면 이렇게 말할 수도 있겠네요.

> 자율적인 객체란 스스로의 의지와 판단에 따라 각자 맡은 책임을 수행하는 객체를 의미한다. 이런 자율적인 객체는 객체지향 공동체를 구성하는 기본 단위이다. 객체지향 설계에서 적절한 책임을 적절한 객체에게 할당해야 한다. 적절한 책임이 자율적인 객체를 낳고, 자율적인 객체들이 모여 유연하고 단순한 협력을 낳는다. 따라서 객체의 자율성이 전체 애플리케이션의 품질을 결정한다. - _객체지향의 사실과 오해, 5장. 책임과 메세지_